client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -195,19 +202,19 @@ const Checkout = React.createClass( {
 			return selectedSiteSlug
 				? `/plans/${ selectedSiteSlug }/thank-you`
 				: '/checkout/thank-you/plans';
-		} else if ( cart.create_new_blog && cartItems.hasDomainRegistration( cart ) && ! cartItems.hasPlan( cart ) ) {
-			const domainName = cartItems.getDomainRegistrations( cart )[ 0 ].meta;
+		} else if ( cart.create_new_blog ) {
+			const domainName = getDomainNameFromReceiptOrCart( receipt, cart );
+			if ( domainName ) {
+				return domainManagementList( domainName );
+			}
 
-			return domainManagementList( domainName );
+			return '/start/domain-first';
```

Comment:
Sooo, I left that for us to decide in the future. I wanted to put something sane there. :) But we can probably default to that page, as it will display the `Ooops, we failed obtaining...`.